Automattic/jetpack
# VideoPress: Allow empty video description and caption

Fixes #26528.

#### Changes proposed in this Pull Request:

* Change the caption and description value checks to allow empty values, but only when the field is really present on the request body

#### Other information:


#### Jetpack product discussion



N/A.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* On your test site, upload one video and get it's attachment ID (or use one that you already uploaded before)
* Do a GET request to `http://your-site.com/wp-json/wp/v2/media/{ID}` to get the video information
* Check the value for the description and caption fields inside the `jetpack_videopress` field
* Now do a POST request to `http://your-site.com/wp-json/wpcom/v2/videopress/meta` with the following `body` to set a new value for the description

```
{
    "id": {ID},
    "description": "Some value"
}
```

* Check the value of the description inside jetpack_videopress field again; the value must be same that you put on the POST request (this was already working before this PR)
* Now do another POST request to `http://your-site.com/wp-json/wpcom/v2/videopress/meta`, but use an empty string for the the description
* Check the value of the description inside jetpack_videopress field again; the value must be empty as you sent (this is the fiexed behavior)
* Do the same for the `caption` field

Some snippets:

**getting the video details**
```cli
curl -X GET -u "user:pass" http://your-site.com/wp-json/wp/v2/media/{ID} | jq
```

**posting the new value for description...**

```cli
curl -X POST -u "user:pass" http://your-site.com/wp-json/wpcom/v2/videopress/meta -H 'Content-Type: application/json' -d '{"id": {ID},"description": "Some value"}' | jq
```

**... and for caption...**
```cli
curl -X POST -u "user:pass" http://your-site.com/wp-json/wpcom/v2/videopress/meta -H 'Content-Type: application/json' -d '{"id": {ID},"caption": "Some value"}' | jq
```

**... or both**
```cli
curl -X POST -u "user:pass" http://your-site.com/wp-json/wpcom/v2/videopress/meta -H 'Content-Type: application/json' -d '{"id": {ID},"caption": "Some value","description": "Some value"}' | jq
```
